influxql/functions.go

```diff
@@ -749,6 +749,16 @@ func ReducePercentile(percentile float64) ReduceFunc {
 	}
 }
 
+// IsNumeric returns whether a given aggregate can only be run on numeric fields.
+func IsNumeric(c *Call) bool {
+	switch c.Name {
+	case "sum", "mean", "median", "min", "max", "spread", "stddev", "percentile":
```

Comment:
I would invert this. We should assume that the aggregate is numeric since it most cases it is. That way the check in planning will get run by default if someone adds a new aggregate function but forgets to add it here.